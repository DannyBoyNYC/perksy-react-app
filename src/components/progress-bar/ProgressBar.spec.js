const React = require('react');

const ProgressBar = require('./index').default;

describe('<ProgressBar.* />', () => {
  let component;
  const requiredProps = {
    current: 100,
    maximum: 1000
  };

  describe('<ProgressBar.Basic />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <ProgressBar.Basic {...requiredProps} />,
        'ProgressBar'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a title if given one', () => {
      component = shallowUntilTarget(
        <ProgressBar.Basic {...requiredProps} title="Title Text" />,
        'ProgressBar'
      );
      expect(component).toIncludeText('Title Text');
    });
  });

  describe('<ProgressBar.Percentage />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <ProgressBar.Percentage {...requiredProps} />,
        'ProgressBar'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render the percentage value', () => {
      expect(component).toIncludeText('10%');
    });
  });
});
