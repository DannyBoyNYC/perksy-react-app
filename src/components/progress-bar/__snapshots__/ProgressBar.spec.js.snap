// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar.* /> <ProgressBar.Basic /> should render correctly with all required props 1`] = `
<div
  className="ProgressBar-root-0-1-1"
>
  <div
    className="ProgressBar-label-0-1-5"
  >
    100 / 1000
  </div>
  <div
    className="ProgressBar-bar-0-1-2"
  >
    <div
      className="ProgressBar-filler-0-1-3"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar.* /> <ProgressBar.Percentage /> should render correctly with all required props 1`] = `
<div
  className="ProgressBar-root-0-1-1"
>
  <div
    className="ProgressBar-label-0-1-5"
  >
    100 / 1000
  </div>
  <div
    className="ProgressBar-bar-0-1-2"
  >
    <div
      className="ProgressBar-filler-0-1-3"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
  <div
    className="ProgressBar-percentageValue-0-1-4"
  >
    10
    %
  </div>
</div>
`;
